subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/conflicts/VersionConflictException.java

```diff
@@ -19,54 +19,81 @@
+        this.resolutions = resolutions;
     }
 
     public List<Pair<List<? extends ModuleVersionIdentifier>, String>> getConflicts() {
         return conflicts;
     }
 
-    private static String buildMessage(String projectPath, String configurationName, Collection<Pair<List<? extends ModuleVersionIdentifier>, String>> conflicts) {
+    private static String buildMessage(Collection<Pair<List<? extends ModuleVersionIdentifier>, String>> conflicts) {
         TreeFormatter formatter = new TreeFormatter();
-        String dependencyNotation = null;
-        int count = 0;
-        formatter.node("Conflict(s) found for the following module(s)");
+
+        int size = conflicts.size();
+        String plural = size > 1 ? "s" : "";
+        formatter.node("Conflict" + plural + " found for the following module" + plural);
         formatter.startChildren();
-        for (Pair<List<? extends ModuleVersionIdentifier>, String> allConflict : conflicts) {
-            if (count > MAX_SEEN_MODULE_COUNT) {
-                formatter.node("... and more");
-                break;
-            }
-            formatter.node(allConflict.right);
-            count++;
-            if (dependencyNotation == null) {
-                ModuleVersionIdentifier identifier = allConflict.getLeft().get(0);
-                dependencyNotation = identifier.getGroup() + ":" + identifier.getName();
-            }
+
+        conflicts.stream().limit(MAX_SEEN_MODULE_COUNT)
+            .forEach(allConflict -> formatter.node(allConflict.right));
+
+        if (size > MAX_SEEN_MODULE_COUNT) {
+            formatter.node("... and more");
         }
         formatter.endChildren();
-        appendInsight(projectPath, configurationName, formatter, dependencyNotation);
         return formatter.toString();
     }
 
-    private static void appendInsight(String projectPath, String configurationName, TreeFormatter formatter, String dependencyNotation) {
+    private static String getDependencyNotation(Collection<Pair<List<? extends ModuleVersionIdentifier>, String>> conflicts) {
+        return conflicts.stream()
+            .findFirst()
+            .map(p -> {
+                ModuleVersionIdentifier identifier = p.left.get(0);
+                return identifier.getGroup() + ":" + identifier.getName();
+            })
+            .orElse(null);
+    }
+
+    private static List<String> createResolutions(String projectPath, String configurationName, String dependencyNotation) {
         if (projectPath.equals(":")) {
             projectPath = "";
         }
-        formatter.node("Run with:");
-        formatter.node("    --scan or");
-        formatter.node("    " + projectPath + ":dependencyInsight --configuration " + configurationName + " --dependency " + dependencyNotation);
-        formatter.node("to get more insight on how to solve the conflict.");
+
+        return ImmutableList.of("Run with: --scan or " + projectPath + ":dependencyInsight --configuration " +
```

Comment:
❓ Is the version conflict failure the only case where the dependency management engine is providing a suggestion?

Reply:
as far as I could find yes